File: baekjoon/1010_s5_다리놓기/다리놓기.py
test = int(input())

def combination (n, m):
  if (n == m):
    return 1
  first = 1
  second = 1
  for i in range(n):
    first = first * (m - i)
  for j in range(n):
    second = second * (n - j)
  return first // second


for _ in range(test):
  n, m = map(int, input().split())
  print(combination(n, m))

File: baekjoon/silver/5_1010_다리놓기/다리놓기.js
//실버5: https://www.acmicpc.net/problem/1010

const fs = require("fs");
const file = process.platform === "linux" ? "dev/stdin" : "./text.txt";
const input = fs.readFileSync(file).toString().trim().split("\n");

input.shift(0);

const combination = (n, k) => {
  let parent = 1;
  let child = 1;

  for (let i = 0; i < k; i++) {
    parent *= n - i;
    child *= k - i;
  }
  console.log(Math.round(parent / child));
};

for (let i = 0; i < input.length; i++) {
  const arr = input[i].split(" ");
  k = parseInt(arr[0]);
  n = parseInt(arr[1]);

  combination(n, k);
}
